pkg/helper/webconsole.go

```diff
@@ -0,0 +1,112 @@
+package helper
+
+import (
+	"context"
+	"fmt"
+	appsv1alpha1 "github.com/3scale/3scale-operator/pkg/apis/apps/v1alpha1"
+	"github.com/RHsyseng/operator-utils/pkg/logs"
+	"github.com/RHsyseng/operator-utils/pkg/resource/read"
+	"github.com/RHsyseng/operator-utils/pkg/utils/kubernetes"
+	consolev1 "github.com/openshift/api/console/v1"
+	routev1 "github.com/openshift/api/route/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+var logger = logs.GetLogger("openshift-webconsole")
+
+// ConsoleLinkText is the text of the consoleLink shown on the webconsole
+const ConsoleLinkText = "APIManager - 3scale"
+
+// ConsoleLinkSupported checks if a ConsoleLink CRD exists
+func ConsoleLinkSupported() error {
+	gvk := schema.GroupVersionKind{Group: "console.openshift.io", Version: "v1", Kind: "ConsoleLink"}
+	return kubernetes.CustomResourceDefinitionExists(gvk)
+}
+
+// CreateConsoleLink creates a ConsoleLink object if it doesn't already exist
+func CreateConsoleLink(ctx context.Context, c client.Client, apimanager *appsv1alpha1.APIManager) {
+	route := getRoute(ctx, c, apimanager)
+	if route != nil {
+		consoleLinkName := fmt.Sprintf("%s-%s", apimanager.ObjectMeta.Name, apimanager.Namespace)
```

Comment:
Try to create `ReconcileConsoleLink` method of the `BaseAPIManagerLogicReconciler`  type in `base_apimanager_logic_reconciler.go`. You should have everything you need and the implementation should be easy.

Reply:
I see the pattern 3scale is using to do the reconciliation, so adding the ReconcileConsoleLink method in that base reconciler, do you mean consolelink creation (as well as the DeleteConsoleLink method) should use this pattern as well?  That means, the desired and/or original Consolelink objects should be constructed at the right time, and eventually are passed to BaseReconciler.ReconcileResource() to determine the creation or update?  After that, the webconsole.go could be removed?

Reply:
new method created